lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -1194,6 +1194,27 @@ struct AIEObjectFifoStatefulTransformPass
     std::set<TileOp>
         objectFifoTiles; // track cores to check for loops during unrolling
 
+    //===------------------------------------------------------------------------===//
+    // Check whether the same data type is used in I/O FIFOs of the objectFifoLink
+    //===------------------------------------------------------------------------===//
+    // We compare the data type of the input and the output FIFOs of the 
+    // objectFifoLink and emits a warning if there is a mismatch
+
+    for(auto linkOp : device.getOps<ObjectFifoLinkOp>()){
+      auto fifoIn = linkOp.getInputObjectFifos()[0];
+      auto fifoOut = linkOp.getOutputObjectFifos()[0];
```

Comment:
There may be multiple fifoIns and fifoOuts  (not at the same time), so for completeness you should check them all.